src/migration/service/migrator.ts

```diff
@@ -57,3 +85,40 @@ export async function resolveSqlMigrations(migrationPath: string): Promise<SqlMi
 
   return Promise.all(migrationPromises);
 }
+
+/**
+ * Resolve all the migration source for each of the migration entries using the configurations.
+ *
+ * @param {string} migrationPath
+ * @param {string} extension
+ * @returns {Promise<SqlMigrationEntry[]>}
+ */
+export async function resolveJavaScriptMigrations(
+  migrationPath: string,
+  extension: string = FileExtensions.JS
+): Promise<JavaScriptMigrationEntry[]> {
+  const migrationNames = await getJavaScriptMigrationNames(migrationPath, extension);
+
+  if (extension === FileExtensions.TS) {
+    // Transpile & execute ts files required on the fly
+    require('ts-node').register({
+      transpileOnly: true
+    });
+  }
+
+  // On the fly es6 => commonJS
+  const esmRequire = require('esm')(module);
+
+  const migrationPromises = migrationNames.map(async name => {
+    const filename = `${name}.${extension}`;
+
+    const { up, down } = esmRequire(path.resolve(migrationPath, filename));
+
+    return {
+      name,
+      queries: { up, down }
```

Comment:
These aren't queries, unlike the sql migrations, so don't send them wrapped in the `queries` object. It is misleading. 
```suggestion
      methods: { up, down }
```